tests/tests/margin_call_fee_tests.cpp

```diff
@@ -0,0 +1,2273 @@
+         // = 600 satoshi CORE / 17 satoshi SMARTBIT
+         //
+         // Expected payment by call order = 2000000 satoshi SMARTBIT * (600 satoshi CORE / 17 satoshi SMARTBIT)
+         // = 2000000 * 600 satoshi CORE / 17
+         // = 70588235.2941 satoshi CORE rounding up to 70588236 satoshi CORE = 705.88236 CORE
+         const asset expected_payment_from_bob_core = core.amount(70588236);
+
+         // Expected fee = payment by call order - payment to limit order
+         // fee = (70588236 - 68235295) satoshi CORE = 2352941 satoshi CORE = 23.52941 CORE
+         const asset expected_margin_call_fee =
+                 expected_payment_from_bob_core - expected_payment_to_alice_core; // core.amount(2352941);
+
+         // Check Alice's balances
+         BOOST_CHECK_EQUAL(get_balance(alice, smartbit), 0);
+         BOOST_CHECK_EQUAL(get_balance(alice, core),
+                           alice_initial_core.amount.value + expected_payment_to_alice_core.amount.value);
+
+         // Check Alice's limit order is closed
+         BOOST_CHECK(!db.find(alice_order_id));
+
+         // Check Bob's debt position is closed
+         BOOST_CHECK(!db.find(bob_call_id));
+
+         // Check Bob's balances
+         // Bob should have no debt asset
+         BOOST_CHECK_EQUAL(get_balance(bob_id(db), smartbit_id(db)), 0);
+         // Bob should have collected the balance of his collateral after the margin call
+         BOOST_CHECK_EQUAL(get_balance(bob_id(db), core_id(db)),
+                           bob_initial_core.amount.value - expected_payment_from_bob_core.amount.value);
+
+         // Check the asset owner's accumulated asset fees
+         BOOST_CHECK(smartbit.dynamic_asset_data_id(db).accumulated_fees == 0);
+         BOOST_CHECK_EQUAL(smartbit.dynamic_asset_data_id(db).accumulated_collateral_fees.value,
+                           expected_margin_call_fee.amount.value);
+
+         // Check the fee of the fill operations for Alice and Bob
+         generate_block(); // To trigger db_notify() and record pending operations into histories
+         graphene::app::history_api hist_api(app);
+         vector <operation_history_object> histories;
+         const int fill_order_op_id = operation::tag<fill_order_operation>::value;
+
+         // Check Alice's history
+         histories = hist_api.get_account_history_operations(
+                 "alice", fill_order_op_id, operation_history_id_type(), operation_history_id_type(), 100);
+         // There should be one fill order operation
+         BOOST_CHECK_EQUAL(histories.size(), 1);
+         // Alice's fill order for her limit order should have zero fee
+         fill_order_operation alice_fill_op = histories.front().op.get<fill_order_operation>();
+         BOOST_CHECK(alice_fill_op.fee == asset(0));
+         // Alice's fill order's fill price should equal the expected match price
+         BOOST_CHECK(~alice_fill_op.fill_price == expected_match_price);
```

Comment:
@abitmore What do you think should be the `fill_price` recorded in `fill_order_operation` under these different maker and taker scenarios?